compiler/src/Language/Bond/Codegen/Cs/Types_cs.hs

```diff
@@ -158,16 +160,18 @@ namespace #{csNamespace}
             else [lt|,
 
             // This class parameters
-            #{commaLineSep 3 thisParam structFields}|]
+            #{commaLineSep 3 paramDeclWithUniqueName (zip structFields uniqueThisFieldNames)}|]
 
         baseFieldList = concat $ baseFields s
 
-        uniqueBaseFieldNames = uniqueNames [fieldName f | f <- baseFieldList]
+        uniqueBaseFieldNames = uniqueNames (map fieldName baseFieldList) []
+        uniqueThisFieldNames = uniqueNames (map fieldName structFields) uniqueBaseFieldNames
 
-        baseParam (f, n) = [lt|#{csType $ fieldType f} #{n}|]
-        thisParam f = [lt|#{csType $ fieldType f} #{uniqueName (fieldName f) uniqueBaseFieldNames}|]
+        paramDecl f = [lt|#{csType $ fieldType f} #{fieldName f}|]
+        paramDeclWithUniqueName (f, n) = [lt|#{csType $ fieldType f} #{n}|]
 
-        paramBasedInitializer f = [lt|this.#{fieldName f} = #{uniqueName (fieldName f) uniqueBaseFieldNames};|]
+        paramBasedInitializer f = [lt|this.#{fieldName f} = #{fieldName f};|]
```

Comment:
Consider merging these two so we always pass a tuple: sometimes it's the normal name and sometimes its the unique name.

Reply:
Addressed in next iteration.